### PR TITLE
ref(deletion): Clean up tests and code for group deletions

### PR DESCRIPTION
**Pull request by armenzg**
Minor changes to help clarify the tests and types.

### CODE REVIEW

File: src/sentry/api/endpoints/project_group_index.py [hunk 1]
```diff
@@ -270,7 +270,12 @@ def delete(self, request: Request, project) -> Response:
         any data mutation.
 
         :qparam int id: a list of IDs of the issues to be removed.  This
-                        parameter shall be repeated for each issue.
+                        parameter shall be repeated for each issue, e.g.
+                        `?id=1&id=2&id=3`. If this parameter is not provided,
+                        it will attempt to remove the first 1000 issues.
+        :qparam querystring query: an optional Sentry structured search
```
Comment (armenzg) [inline src/sentry/api/endpoints/project_group_index.py:276]: If you look at the bulk delete tests you will see that it is being used.
You can see this documented in the get method:
https://github.com/getsentry/sentry/blob/90006fc294c1160979416ca11cda2c75846af8b1/src/sentry/api/endpoints/project_group_index.py#L75-L77

It is likely that more options can be passed and documented, however, I will leave that for another pull request.

File: tests/snuba/api/endpoints/test_project_group_index.py [hunk 2]
```diff
@@ -1429,15 +1436,11 @@ def test_delete_by_id(self, mock_eventstream):
...
 
-        hashes = []
-        for g in group1, group2, group3, group4:
-            hash = uuid4().hex
-            hashes.append(hash)
-            GroupHash.objects.create(project=g.project, hash=hash, group=g)
+        self.create_associated_hashes([group1, group2, group3, group4])
 
         self.login_as(user=self.user)
-        url = f"{self.path}?id={group1.id}&id={group2.id}&group4={group4.id}"
```
Comment (armenzg) [inline tests/snuba/api/endpoints/test_project_group_index.py:1439]: Notice `group4=` in the line deleted.

```diff
-         url = f"{self.path}?id={group1.id}&id={group2.id}&group4={group4.id}"
+         url = f"{self.path}?id={group1.id}&id={group2.id}&id={group4.id}"
```

File: tests/snuba/api/endpoints/test_project_group_index.py [hunk 3]
```diff
@@ -1446,18 +1449,17 @@ def test_delete_by_id(self, mock_eventstream):
...
-        assert Group.objects.get(id=group1.id).status == GroupStatus.PENDING_DELETION
-        assert not GroupHash.objects.filter(group_id=group1.id).exists()
-
-        assert Group.objects.get(id=group2.id).status == GroupStatus.PENDING_DELETION
-        assert not GroupHash.objects.filter(group_id=group2.id).exists()
-
-        assert Group.objects.get(id=group3.id).status != GroupStatus.PENDING_DELETION
-        assert GroupHash.objects.filter(group_id=group3.id).exists()
+        # Groups pending for deletion
+        for g in [group1, group2]:
```
Comment (armenzg) [inline tests/snuba/api/endpoints/test_project_group_index.py:1453]: I'm grouping the code into loops.

File: tests/snuba/api/endpoints/test_project_group_index.py [hunk 4]
```diff
@@ -1446,18 +1449,17 @@ def test_delete_by_id(self, mock_eventstream):
...
+            assert Group.objects.get(id=g.id).status == GroupStatus.PENDING_DELETION
+            assert not GroupHash.objects.filter(group_id=g.id).exists()
 
-        assert Group.objects.get(id=group4.id).status != GroupStatus.PENDING_DELETION
-        assert GroupHash.objects.filter(group_id=group4.id).exists()
+        # Groups not deleted
+        for g in [group3, group4]:
```
Comment (armenzg) [inline tests/snuba/api/endpoints/test_project_group_index.py:1458]: I'm grouping the code into loops.

File: tests/snuba/api/endpoints/test_project_group_index.py [hunk 5]
```diff
@@ -1473,17 +1475,15 @@ def test_delete_by_id(self, mock_eventstream):
...
-        assert not Group.objects.filter(id=group1.id).exists()
-        assert not GroupHash.objects.filter(group_id=group1.id).exists()
-
-        assert not Group.objects.filter(id=group2.id).exists()
-        assert not GroupHash.objects.filter(group_id=group2.id).exists()
-
-        assert Group.objects.filter(id=group3.id).exists()
-        assert GroupHash.objects.filter(group_id=group3.id).exists()
+        # Completely deleted groups
+        for g in [group1, group2]:
```
Comment (armenzg) [inline tests/snuba/api/endpoints/test_project_group_index.py:1479]: I'm grouping the code into loops.

File: tests/snuba/api/endpoints/test_project_group_index.py [hunk 6]
```diff
@@ -1473,17 +1475,15 @@ def test_delete_by_id(self, mock_eventstream):
...
+            assert not Group.objects.filter(id=g.id).exists()
+            assert not GroupHash.objects.filter(group_id=g.id).exists()
 
-        assert Group.objects.filter(id=group4.id).exists()
-        assert GroupHash.objects.filter(group_id=group4.id).exists()
+        # Groups not deleted
+        for g in [group3, group4]:
```
Comment (armenzg) [inline tests/snuba/api/endpoints/test_project_group_index.py:1484]: I'm grouping the code into loops.

File: tests/snuba/api/endpoints/test_project_group_index.py [hunk 7]
```diff
@@ -1568,17 +1560,14 @@ def test_bulk_delete_performance_issues(self):
...
-            hashes.append(hash)
-            GroupHash.objects.create(project=group.project, hash=hash, group=group)
+        self.create_associated_hashes(groups)
 
         self.login_as(user=self.user)
 
         # if query is '' it defaults to is:unresolved
         url = self.path + "?query="
         response = self.client.delete(url, format="json")
+        assert response.json() == ["Only error issues can be deleted."]
```
Comment (armenzg) [inline tests/snuba/api/endpoints/test_project_group_index.py:1570]: I'm adding this to be clear why we get a 400.

**Review by JoshFerge — APPROVED** [review]

**Pull request merged**